packages/block/src/header.ts

```diff
@@ -135,6 +135,15 @@ export class BlockHeader {
       throw new Error('invalid header. Less values than expected were received')
     }
 
+    // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+    if (opts.common?.isActivatedEIP(1559) && baseFeePerGas === undefined) {
+      const eip1559ActivationBlock = bigIntToBuffer(opts.common?.eipBlock(1559)!)
+      // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+      if (eip1559ActivationBlock && eip1559ActivationBlock.equals(number)) {
+        throw new Error('invalid header. baseFeePerGas should be provided')
+      }
+    }
```

Comment:
I find this code part extremely hacky (only for this one activation block, check only in the value constructor, endless linting rule exceptions).

Is this really necessary in the current form, or can we at least improve here?